src/domain/usecases/ValidateOptionSetsUseCase.ts

```diff
@@ -37,66 +36,69 @@ export class ValidateOptionSetsUseCase {
             const namingConventionRules = this.validateNamingConventions(option);
             const misplacedCommasRules = this.validateMisplacedCommas(option);
             const orderRules = this.validateOrder(option, optionIndex);
-            return {
-                option,
-                optionSet,
-                rules: [
-                    ...invalidLengthRules,
-                    ...namingConventionRules,
-                    ...misplacedCommasRules,
-                    ...orderRules,
-                ],
-            };
+
+            const allOptionsErrors = _([
+                invalidLengthRules,
+                namingConventionRules,
+                misplacedCommasRules,
+                orderRules,
+            ])
+                .compact()
+                .value();
+
+            return { option, optionSet, errors: allOptionsErrors };
         });
     }
 
-    private validateLength(option: Option, validationLength: number): OptionValidationRule[] {
+    private validateLength(option: Option, validationLength: number): Maybe<OptionValidationError> {
         return option.code.length > validationLength
-            ? [{ type: "invalid_length", message: `Code length is ${option.code.length}` }]
-            : [];
+            ? { type: "invalid_length", message: `Code length is ${option.code.length}` }
+            : undefined;
     }
 
-    private validateNamingConventions(option: Option): OptionValidationRule[] {
+    private validateNamingConventions(option: Option): Maybe<OptionValidationError> {
         const hasSpaces = /\s/.test(option.code);
-        const hasSpecialCharacters = option.code.match(/[^\w\s]/) ?? [];
+        const specialCharacters = option.code.match(/[^\w\s,]/g) ?? [];
 
-        const spacesRule = hasSpaces ? "spaces" : "";
+        const spacesRule = hasSpaces ? "spaces" : undefined;
         const specialRule =
-            hasSpecialCharacters.length > 0 ? `special characters: ${hasSpecialCharacters.join(",")}` : "";
+            specialCharacters.length > 0 ? `special characters: ${specialCharacters.join(",")}` : undefined;
 
-        const message = [spacesRule, specialRule].filter(Boolean).join(" and ");
+        const message = _([spacesRule, specialRule]).compact().join(" and ");
 
         return spacesRule || specialRule
-            ? [{ type: "naming_conventions", message: `Code contains ${message}` }]
-            : [];
+            ? { type: "naming_conventions", message: `Code contains ${message}` }
+            : undefined;
     }
 
-    private validateMisplacedCommas(option: Option): OptionValidationRule[] {
-        return option.code.includes(",")
-            ? [{ type: "misplaced_commas", message: `Code has misplaced commas` }]
-            : [];
+    private validateMisplacedCommas(option: Option): Maybe<OptionValidationError> {
+        const hasMisplacedCommas = /^,|,,|,$|,/.test(option.code);
```

Comment:
Regexp are not declarative enough, so we'd need a comment of what it's considered a misplaced comma. 

It's weird, because we have at-start, double, at-end, but also ",", which is any comma, right? so the previous checks are not useful.